addons/purchase_stock/models/product.py

```diff
@@ -11,10 +11,9 @@ class ProductTemplate(models.Model):
 
     @api.model
     def _get_buy_route(self):
-        buy_route_id = self.env.ref('purchase_stock.route_warehouse0_buy', raise_if_not_found=False).id
-        buy_route = self.env['stock.location.route'].search([('id', '=', buy_route_id)])
+        buy_route = self.env.ref('purchase_stock.route_warehouse0_buy', raise_if_not_found=False)
         if buy_route:
-            return buy_route.ids
+            return self.env['stock.location.route'].search([('id', '=', buy_route.id)]).ids
```

Comment:
return buy_route.ids

Reply:
`self.env['stock.location.route'].search([('id', '=', buy_route.id)])` was made on this commit: https://github.com/odoo/odoo/pull/88688

You even recommended this search, do you want to get rid of it now ?

Reply:
Doing this will break the use case of this PR https://github.com/odoo/odoo/pull/88688 and return the code to its previous state (https://github.com/odoo/odoo/blob/57919cec81fb65b7e8774cce7cfa079793e53138/addons/purchase_stock/models/product.py)

Reply:
Good point, I forgot that stuff